tests/thread_float/tests/01-run.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+#  Copyright (C) 2021 Otto-von-Guericke-Universität Magdeburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+# @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+
+import sys
+from testrunner import run
+
+
+def testfunc(child):
+    child.expect("THREADS CREATED")
+    child.expect("")
+    child.expect("THREAD t1 start")
+    child.expect("THREAD t2 start")
+    child.expect("THREAD t3 start")
+
+    child.expect(r"t(\d): (\d{3}\.\d+)")
+    first_thread = int(child.match.group(1))
+    first_result = float(child.match.group(2))
+    assert (first_result > 141.4) and (first_result < 141.5), f"result of t{first_thread} wrong"
+    child.expect(r"t(\d): (\d{3}\.\d+)")
+    second_thread = int(child.match.group(1))
+    second_result = float(child.match.group(2))
+    assert (second_result > 141.4) and (second_result < 141.5), f"result of t{second_thread} wrong"
+    assert (first_thread == 1) or (first_thread == 3), "only thread t1 and t3 should print"
+    assert (second_thread == 1) or (second_thread == 3), "only thread t1 and t3 should print"
+    assert first_thread != second_thread, "t1 and t3 should print in turns"
+    for _ in range(3):
+        child.expect(r"t(\d): (\d{3}\.\d+)")
+        assert first_thread == int(child.match.group(1)), "t1 and t3 should print in turns"
+        assert first_result == float(child.match.group(2)), "same calculation but different result"
```

Comment:
Still not sure if `==` comparison with floats is a good idea in tests… But let's fix things when they actually become a problem.

Reply:
I mean... there is a reason, why Python's `unittest` module offers a [`assertAlmostEqual()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual) method ;-) (not that I say you should use that here, though unittest-style tests are [indeed possible](https://github.com/RIOT-OS/RIOT/blob/master/dist/pythonlibs/testrunner/unittest.py)).

Reply:
I agree that the same calculation on different hardware can yield different results, or the same calculation on the same hardware just calculate in a different but mathematical equivalent way can yield different results. That's one if the pain points of floats IMO.

But the exact same calculation done using the exact same steps on the exact same FPU should still yield the same result bit for bit. After all, FPUs are fully deterministic.

Reply:
But, technically, you are not comparing the result of the calculations here, but whatever Python thinks it should cast the strings that were generated from these calculations results into. So maybe, to reduce points of failure, it would be best to compare the strings directly, not the casted floats after all. Will provide suggestions for that.

Reply:
```suggestion
        assert first_result == child.match.group(2), "same calculation but different result"
```

Reply:
fixed